### PR TITLE
Unfiltered "unknown" countries from the geo data.

### PR DESCRIPTION
**Pull request by dsjen**
@rocha @clintonb 

FYI: @lamagnifica @shnayder

### CODE REVIEW

**Comment by rocha** [comment]
A test is failing and it looks related, can you check?

**Comment by dsjen** [comment]
Fixed!

**Comment by shnayder** [comment]
This would still display "UNKNOWN" in the top 3 if it's one of the top 3 counts, right?

**Comment by clintonb** [comment]
@shnayder Yes. According to [AN-3494](https://openedx.atlassian.net/browse/AN-3494) that is not the desired behavior.

**Comment by dsjen** [comment]
Yes, I was going to submit another PR that filtered this out of the top three.  The goal of this PR is to put the unknown countries in the data table, which we thought was the current behavior.

I'm working on another PR that removes the unknown from the top three.

**Comment by clintonb** [comment]
Let's just combine the two changes since they cover the same area of code.

**Comment by dsjen** [comment]
Updated:

screen shot 2014-09-30 at 10 56 56 am

**Comment by shnayder** [comment]
Looks good to me.

**Comment by clintonb** [comment]
@dsjen You'll need to rebase on master to pick up the Travis config.

**Comment by lamagnifica** [comment]
@clintonb please do not forget to change the labels on these tiles!

File: analytics_dashboard/courses/presenters.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ class CourseEnrollmentPresenter(BasePresenter):
     """ Presenter for the course enrollment data. """
 
     NUMBER_TOP_COUNTRIES = 3
+    UNKNOWN_COUNTRY_NAME = 'UNKNOWN'
```
Comment (clintonb) [inline analytics_dashboard/courses/presenters.py]: Use `analyticsapiclient.constants.UNKNOWN_COUNTRY_CODE`.

File: analytics_dashboard/courses/tests/utils.py [hunk 2]
```diff
@@ -62,11 +62,12 @@ def get_mock_presenter_enrollment_geography_data():
         {'countryCode': 'USA', 'countryName': 'United States', 'count': 500, 'percent': 0.5},
         {'countryCode': 'CAN', 'countryName': 'Canada', 'count': 300, 'percent': 0.3},
         {'countryCode': 'GER', 'countryName': 'Germany', 'count': 100, 'percent': 0.1},
+        {'countryCode': None, 'countryName': 'UNKNOWN', 'count': 100, 'percent': 0.1},
```
Comment (clintonb) [inline analytics_dashboard/courses/tests/utils.py]: You can test the filtering by making UNKOWN one of the top 3 (e.g. swap the count and percentage with Canada) and ensuring it does not appear in `top_countries` below.

**Comment by dsjen** [comment]
Rebase and updated tests to show unknown countries filtered from summary.

File: analytics_dashboard/courses/tests/utils.py [hunk 3]
```diff
@@ -60,15 +61,18 @@ def get_mock_api_enrollment_geography_data_limited(course_id):
...
+        {'countryCode': None, 'countryName': UNKNOWN_COUNTRY_CODE, 'count': 300, 'percent': 0.3},
     ]
     summary = {
         'last_updated': CREATED_DATETIME,
         'num_countries': 3,
-        'top_countries': data
+        'top_countries': data[:3]
     }
 
+    data = sorted(data, key=lambda i: i['count'], reverse=True)
```
Comment (clintonb) [inline analytics_dashboard/courses/tests/utils.py:76]: You can avoid this call by simply moving the unknown entry below USA.

**Comment by clintonb** [comment]
:+1:  After the sorting is fixed.

**Comment by lamagnifica** [comment]
@dsjen I am concerned about the labels on these tiles...

**Comment by rocha** [comment]
@lamagnifica what do you suggest?

**Comment by clintonb** [comment]
@lamagnifica Labels will be fixed in AN-3557. This PR is meant to simply add the unknown entry to the table and keep it out of the tiles.

File: analytics_dashboard/courses/tests/utils.py (hunk 3 above)
Comment (dsjen) [inline analytics_dashboard/courses/tests/utils.py:76]: If the unknown entry is reordered, the `top_countries` need to be sliced different (currently `'top_countries': data[:3]`.  Sorting seemed cleaner than having to either filter or manually create the `top_countries` listing.

Comment (clintonb) [inline analytics_dashboard/courses/tests/utils.py:76]: Either do the filtering or include a comment regarding why the sorting is necessary.

**Comment by dsjen** [comment]
@lamagnifica is it ok with you if we fix the labels in AN-3557?

File: analytics_dashboard/courses/tests/utils.py (hunk 3 above)
Comment (dsjen) [inline analytics_dashboard/courses/tests/utils.py:76]: Commented code.

**Comment by lamagnifica** [comment]
when will AN-3557 be released?

**Comment by lamagnifica** [comment]
@rocha :https://openedx.atlassian.net/wiki/display/AN/Labels+and+Tooltips+-+Insights

**Comment by clintonb** [comment]
@lamagnifica No one has yet started work on AN-3557, so I cannot say for certain.

**Comment by dsjen** [comment]
AN-3557 isn't considered a release blocker, and we're focusing on these currently.  However, the copy changes look easy enough, and I can incorporate them in this PR (unless anyone objects).

**Comment by clintonb** [comment]
@dsjen I object. We already have a plan to address the copy changes. Resolving the unknown countries takes higher priority. This is ready to merge now. Let's merge it and address copy changes in another PR.

**Comment by dsjen** [comment]
@lamagnifica I went ahead and merged this PR.  We can submit another PR for the copy changes depending on priorities--please let us know (either AN-3557 can be prioritized or the copy changes for the geo summary cards themselves can be).

**Pull request merged**